pokemongo_bot/api_wrapper.py

```diff
@@ -0,0 +1,55 @@
+# api_wrapper.py
+
+from pgoapi import PGoApi
+from pgoapi.exceptions import NotLoggedInException
+import logger
+
+class ApiWrapper(object):
+    def __init__(self, api):
+        self._api = api
+        self.reset_auth()
+
+    def reset_auth(self):
+        self._api._auth_token = None
+        self._api._auth_provider = None
+        self._api._api_endpoint = None
+
+    # wrap api methods
+    def _can_call(self):
+        if not self._api._req_method_list or len(self._api._req_method_list) == 0:
+            raise RuntimeError('Trying to call the api without setting any request')
+        if self._api._auth_provider is None or not self._api._auth_provider.is_login():
+            logger.log('Not logged in!', 'red')
+            raise NotLoggedInException()
+        return True
+
+    def call(self):
+        if not self._can_call():
+            return False
+
+        api_req_method_list = self._api._req_method_list
+        result = None
+        max_retry = 5 # TODO configuration param
+        try_cnt = 0
+        while True:
+            self._api._req_method_list = [req_method for req_method in api_req_method_list] # api internally clear this field after a call
+            result = self._api.call()
```

Comment:
You're right, api.call use request.request so I could check for those exceptions